docs/pages/benchmarking_and_the_dev_process.md

```diff
@@ -0,0 +1,50 @@
+# Benchmarking and the development process
+
+Using benchmarks as a development tool is all about the context those benchmarks come from and are run in. This context is important not only to help interpret the benchmarks (did performance increase or decrease, compared to what?) they also make up the history of measurement for the benchmarks we are running.
+
+Although conbench can store benchmarks from anywhere within a git tree (or even software that is outside of a git tree), we typically benchmark at specific points within a git tree for different purposes. Further, we use the git history to situate where a benchmark came from in history, what came before it, and its relationship to various points of interest in the git tree.
```

Comment:
We want a capital C in Conbench, yes?

I love this paragraph! But maybe we can make it more precise?

Something like "A benchmark result submitted to Conbench is typically associated with a specific commit in a specific git code repository, i.e. a node in a git tree (although one can also store results not associated with any code repository). If commit information is provided with a benchmark result, Conbench uses the git history to ..."

Just some ideas. Feel free to use some or none!

Reply:
I'm on the fence if we cant to lean into the hipper `conbench` versus Conbench, but for now I've gone through and capitalized. 

And yes, I like this more specific | pedantic framing